iceoryx_posh/include/iceoryx_posh/runtime/posh_runtime.hpp

```diff
@@ -83,10 +83,10 @@ class PoshRuntime
 
     /// @brief find all services that match the provided service description
     /// @param[in] serviceDescription service to search for
-    /// @return cxx::expected<InstanceContainer, FindServiceError>
-    /// InstanceContainer: on success, container that is filled with all matching instances
+    /// @return cxx::expected<ServiceContainer, FindServiceError>
+    /// ServiceContainer: on success, container that is filled with all matching service descriptions
     /// FindServiceError: if any, encountered during the operation
-    virtual cxx::expected<InstanceContainer, FindServiceError>
+    virtual cxx::expected<ServiceContainer, FindServiceError>
```

Comment:
> It is very easy to increase the instance container size to iox::MAX_NUMBER_OF_SERVICES and then this can never happen.

Indeed, will do.

> Otherwise we increase again the error handling in unnecessary cases and it would be inconsistent with the behavior of the other roudi communicating methods.

I'm tempted to disagree here. This class will be used quite close to the user-facing API, so we need to return the errors to the user in most case. Will have a thought about this.